src/TTextEdit.cpp

```diff
@@ -832,22 +835,27 @@ void TTextEdit::swap(QPoint& p1, QPoint& p2)
     p2 = tmp;
 }
 
+void TTextEdit::normaliseSelection()
+{
+    if (mPA.y() > mPB.y() || ((mPA.y() == mPB.y()) && (mPA.x() > mPB.x()))) {
+        swap(mPA, mPB);
+    }
+}
+
 void TTextEdit::mouseMoveEvent(QMouseEvent* event)
 {
     if (mFontWidth == 0 || mFontHeight == 0) {
         return;
     }
 
-    int y = (event->y() / mFontHeight) + imageTopLine();
-    y = std::max(y, 0);
-
-    int x = convertMouseXToBufferX(event->x(), y);
+    int lineIndex = std::max(0, (event->y() / mFontHeight) + imageTopLine());
+    int tCharIndex = convertMouseXToBufferX(event->x(), lineIndex);
```

Comment:
Generally speaking it is indexing through the `TChar`s in `(std::deque<std::deque<TChar>>) TBuffer::buffer` (though sometimes also the `QChar`s in `(QStringList) TBuffer::lineBuffer` - and I wanted to avoid a variable name beginning with `ch` or `char` to avoid an IDE autocomplete wanting to make it into the `char` type - and other developers thinking that it is indexing into an array of the `char` type!